BH P100a whisper_performance demo test hangs on first iteration with harvested fw+sw

https://github.com/tenstorrent/tt-metal/actions/runs/14653158459/job/41159933205#step:8:199 
p100a VM
FW ver 80.18.0.0
TT-KMD 1.33
Running blackhole post commit on `williamly/bh-virtual` branch with Almeet's changes to enable noc translation, which is required to enable fw ver 80.18.0.0 on VMs with harvested p100a/p150a cards.

Model hangs on first iteration on a harvested p100a (not seen in p150a)
```
models/demos/whisper/demo/demo.py::test_demo_for_conditional_generation[device_params0-True-2-models.demos.whisper.tt.ttnn_optimized_functional_whisper]                   Metal | INFO     | Initializing device 0. Program cache is NOT enabled
           BuildKernels | INFO     | Skipping deleting built cache
                  Metal | INFO     | AI CLK for device 0 is:   1350 MHz
Special tokens have been added in the vocabulary, make sure the associated word embeddings are fine-tuned or trained.
2025-04-24 23:08:55.633 | WARNING  | ttnn.model_preprocessing:from_torch:529 - ttnn: model cache can be enabled by passing model_name argument to preprocess_model[_parameters] and setting env variable TTNN_CONFIG_OVERRIDES='{"enable_model_cache": true}'
2025-04-24 23:08:57.989 | INFO     | models.demos.whisper.tt.ttnn_optimized_functional_whisper:init_kv_cache:34 - Initializing KV cache with max batch size: 1 and max sequence length: 512
2025-04-24 23:08:58.041 | INFO     | models.demos.whisper.demo.demo:_model_pipeline:249 - Running model on audio data with duration 6.960s
2025-04-24 23:09:12.209 | INFO     | models.demos.whisper.demo.demo:run_generate:122 - Time to encoder states: 14168.323ms

Decode inference iterations:   0%|          | 0/448 [00:00<?, ?it/s]
Decode inference iterations:   0%|          | 1/448 [00:12<1:33:51, 12.60s/it]
Error: The action has timed out.
```

@skhorasganiTT can you take a look? I can provide access to a p100a VM for debug.
Please re-assign if needed as well, thank you.

are there any env vars needed to run this test or flags to use dummy weights? I am seeing:
```
    def load_input_paths(folder_path):
>       files = [os.path.join(folder_path, f) for f in listdir(folder_path) if isfile(join(folder_path, f))]
E       FileNotFoundError: [Errno 2] No such file or directory: ''
```

also @williamlyTT could you provide me access to a p100a?

--- a/tt_metal/hw/inc/dataflow_api.h
+++ b/tt_metal/hw/inc/dataflow_api.h
@@ -377,6 +377,7 @@ void cb_reserve_back(int32_t operand, int32_t num_pages) {
     do {
         // uint16_t's here because Tensix updates the val at tiles_acked_ptr as uint16 in llk_pop_tiles
         // TODO: I think we could have TRISC update tiles_acked_ptr, and we wouldn't need uint16 here
+        invalidate_l1_cache();
         uint16_t pages_acked = (uint16_t)reg_read(pages_acked_ptr);
         uint16_t free_space_pages_wrap =
             get_local_cb_interface(operand).fifo_num_pages - (pages_received - pages_acked);

